### PR TITLE
🦺 #381 - improve invite validation

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI
- unregister guardian fields with `shouldUnregister: true`
- toggle submit button based on form & recaptcha valid status
- change invite form validation from `onSubmit` to `onBlur`
- change checkbox validation to `onChange`

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [ ] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx [hunk 1]
```diff
@@ -49,41 +49,50 @@ const SelectInput = <T extends FieldValues, V extends string>({
...
+						name={name}
+						onBlur={handleBlur}
+						onChange={(val) => {
+							// in react-select the value can be a string or object.
+							// in our implementation it must be {label, value},
+							// with the label being translated.
+							// empty string case doesn't happen in our implementation,
+							// because the user can't unselect this field after
+							// making a selection.
+							// the default unselected value is `undefined`
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx:78]: corrected the comment here

File: apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx [hunk 2]
```diff
@@ -49,41 +49,50 @@ const SelectInput = <T extends FieldValues, V extends string>({
...
-					onFocus={onFocus}
-					openMenuOnFocus
-					options={options}
-					placeholder={placeholder}
-					value={options.find((option) => option.value === value) || ''}
-					{...ariaProps}
-				/>
-			)}
+			render={({ field: { onBlur: controllerOnBlur, onChange, value } }) => {
+				const handleBlur = () => {
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx:53]: added handleBlur function to use the form's blur function (for validation) and the blur function passed down as a prop (for the tooltip)

File: apps/consent-ui/src/components/common/Form/fieldsets/CheckboxFieldSet/index.tsx [hunk 3]
```diff
@@ -49,6 +49,26 @@ const CheckboxFieldSet = <T extends FieldValues>({
 	title,
 }: CheckBoxFieldSetProps<T>) => {
 	const checkboxId = useId();
+
+	// validate onChange
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/CheckboxFieldSet/index.tsx:53]: we'll have to add this to radios too

I can't specify at top-level which validation mode to use for each field

**Review by anncatton — COMMENTED** [review]
The behaviour on this is very nice. Thanks for the helpful comments!

One thing I did notice, if I trigger an error on the `consentToBeContacted` checkbox before I ever touch it, it often requires an extra click to toggle it to checked. Have you encountered this?

**Comment by mistryrn** [comment]
Still reviewing the code changes here, but upon local testing it looks good! Great job updating the validation behaviour.

Only thing I noticed during testing, and I don't remember if we have a follow-up ticket to address this or not, but the regex/validation for the "Preferred Name" field is still preventing form submission if you don't provide a value for it. We're also still only passing the "required" error message to it, even if the error isn't that the field is required.

https://github.com/OHCRN/platform/blob/45a1bd08ac3631f920aa3ce35ee9dac2d3c881fd/apps/consent-ui/src/components/views/Invite/ClinicianInviteForm/index.tsx#L182

Removing the `&& errorsDict.required` from the error prop reveals that the error is `invalid_string` if you don't provide a value to the optional field here:
Screenshot 2024-01-23 at 4 32 08 PM

**Comment by samrichca** [comment]
@mistryrn we have tickets for [improving regex validation](https://github.com/OHCRN/platform/issues/285) and [form error handling](https://github.com/OHCRN/platform/issues/315)

**Comment by samrichca** [comment]
added section heading comments for consistency with register form PR

**Pull request merged**